DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -201,5 +196,12 @@ string password
 
             return loginService.VerifyUsers(password, adminUser, delegateUsers);
         }
+
+        public bool IsPasswordValid(int? adminId, int? delegateId, string password)
+        {
+            var verifiedLinkedUsersAccounts = GetVerifiedLinkedUsersAccounts(adminId, delegateId, password);
+
+            return verifiedLinkedUsersAccounts.Any();
+        }

```

Comment:
This new method makes sense, but it seems to make a call to GetVerifiedLinkedUsersAccounts, which is also performed in the UpdateUserAccountDetails method (called from MyAccountController) and in the ChangePasswordController's Post Index method, so it feels like these methods could be refactored to reduce potentially redundant calls. 

Off the top of my head, it feels like it will be more complicated than it seems. Do you reckon its worth investigating?

Reply:
There isn't really a sensible way to do this without the duplicate calls. We don't want to be going anywhere near the actual bulk of the logic if there are other validation error so we need a basic password check before the ModelState.IsValid check.

Reply:
Fair enough. Was just double checking that it had been considered and dismissed for a good reason.